product-service/src/functions/getProductById/__tests__/getProductById.test.ts

```diff
@@ -0,0 +1,38 @@
+import { getProductById } from '../handler';
+
+describe('Test products/{productId} url', () => {
+  const productList = [
+    {
+      id: '1',
+      name: 'product 1',
+    },
+    {
+      id: '2',
+      name: 'product 2',
+    },
+  ];
+
+  const pathParameters = {
+    productId: '2',
+  };
+
+  test('Should return product by giving productId', () => {
+    console.log(getProductById({ productList, pathParameters }));
+    const { statusCode, body } = getProductById({ productList, pathParameters });
+    expect(statusCode).toBe(200);
+    expect(body).toEqual('{"id":"2","name":"product 2"}');
+  });
+
+  test('Should return 404 when no productList was provided', () => {
+    const { statusCode, body } = getProductById({ productList: undefined, pathParameters });
+    expect(statusCode).toBe(404);
+    expect(body).toBe(undefined);
+  });
+
+  test('Should return 404 with incorrect productId', () => {
+    const incorrectPathParameters = { productId: '7' };
+    const { statusCode, body } = getProductById({ productList, pathParameters: incorrectPathParameters });
+    expect(statusCode).toBe(404);
+    expect(body).toBe(undefined);
+  });
+});
```

Comment:
There is a good practice to have contract tests to validate API response against JSON Schema. I use [jsonschema](https://github.com/tdegrunt/jsonschema) package for that on my current project. It's a recommendation and you are not required to implement that.

Reply:
We use [ajv](https://ajv.js.org/) in our project, on the frontend. I think it should be connected with reusing Open API models. I try and consider this option someday.

Reply:
Yup, good choice. I had experience with ajv as well and it was quite positive.